### PR TITLE
fix(input): fix listening value property changes

### PR DESCRIPTION
**Pull request by muratcorlu**
We realized that our input and textarea components don't listen to changes on value **property** to update their validation states. This PR solves this issue.

While solving the issue, some tests of the pagination component failed. Then I noticed some problems with how we fire and listen to `bl-change` events of pagination. So this PR also includes some refactor in the pagination component.

### CODE REVIEW

File: src/components/input/bl-input.ts [hunk 1]
```diff
@@ -191,22 +191,20 @@ export default class BlInput extends FormControlMixin(LitElement) {
...
 
-  valueChangedCallback(value: string): void {
-    this.value = value;
-  }
-
   private inputHandler(event: Event) {
     const value = (event.target as HTMLInputElement).value;
 
-    this.setValue(value);
+    console.log('inputHandler', value);
```
Comment (AykutSarac) [inline src/components/input/bl-input.ts:197]: I think it's supposed to be removed?

File: src/components/pagination/bl-pagination.test.ts [hunk 2]
```diff
@@ -221,7 +221,7 @@ describe('bl-pagination', () => {
         jumper.value = '5';
       }
 
-      setTimeout(() => jumper?.dispatchEvent(new Event('change')));
+      setTimeout(() => jumper?.dispatchEvent(new Event('change', { bubbles: true })));
```
Comment (AykutSarac) [inline src/components/pagination/bl-pagination.test.ts:224]: Move this line under of `jumper.value`, this will likely to solve the test error.

**Review by AykutSarac — CHANGES_REQUESTED** [review]

File: src/components/pagination/bl-pagination.test.ts (hunk 2 above)
Comment (muratcorlu) [reply src/components/pagination/bl-pagination.test.ts:224]: Do you mean like this?

```ts
      if (jumper) {
        jumper.value = '5';

        setTimeout(() => jumper.dispatchEvent(new Event('change', { bubbles: true })));
      }
```

This doesn't change anything.

Comment (muratcorlu) [reply src/components/pagination/bl-pagination.test.ts]: Indeed, the way we fire change event in pagination was a bit problemmatic. Also input and pagination both uses `bl-change` name and that also creates a confusion. I made some refactors in pagination and added `stopPropogation` for `bl-change` event of internal input to not fire 2 different type of `bl-change` event from pagination component. Please check latest version.

**Review by AykutSarac — APPROVED** [review]

**Review by pratikgaloria — APPROVED** [review]

**Pull request merged**